server/datastore/mysql/software.go

```diff
@@ -1402,8 +1408,17 @@ func (ds *Datastore) InsertSoftwareVulnerability(
 
 	var args []interface{}
 
-	stmt := `INSERT INTO software_cve (cve, source, software_id) VALUES (?,?,?) ON DUPLICATE KEY UPDATE updated_at=?`
-	args = append(args, vuln.CVE, source, vuln.SoftwareID, time.Now().UTC())
+	stmt := `
+		INSERT INTO software_cve (cve, source, software_id, resolved_in_version) 
+		VALUES (?,?,?,?) 
+		ON DUPLICATE KEY UPDATE
+			cve = VALUES(cve),
+			source = VALUES(source),
+			software_id = VALUES(software_id),
+			resolved_in_version = VALUES(resolved_in_version),
+			updated_at=?
+	`
+	args = append(args, vuln.CVE, source, vuln.SoftwareID, vuln.ResolvedInVersion, time.Now().UTC())
 
```

Comment:
changed this to an upsert, any issues with this if the customer has many CVEs?

Reply:
I'm not following, it already was an upsert.

Also, why are we updating these other fields now?
- `software_id`+`cve` are already a `UNIQUE KEY` (so if there's a duplicate, assigning them it's a no-op)
- Why update `source` and `resolved_in_version`?

Reply:
> but we need a way to add the resolved_in_version if it doesn't exist.

Gotcha. I'm ok with updating `source` and `resolved_in_version`.
Do we really need to add `software_id = VALUES(software_id)` and `cve = VALUES(cve)`?

Reply:
not with the new constraints, updated